pkg/main.go

```diff
@@ -18,9 +21,16 @@ func main() {
 	ds.EnableMultipleConnections = true
 	ds.CustomRoutes = routes.New(s).Routes()
 
+	// newDatasourceForUpgradedPluginSdk adds context to the NewDatasource function, which is the new signature expected
+	// for an InstanceFactoryFunc in  grafana-plugin-sdk-go v0.177.0. Moving forward, we can update the NewDatasource
+	// signature in grafana-aws-sdk once all dependent datasources upgrade their grafana-plugin-sdk-go.
+	newDatasourceForUpgradedPluginSdk := func(ctx context.Context, settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
```

Comment:
1. Here I have created the signature expected by the InstanceFactoryFunc type. 
2. Another way we could move forward is to add a new method like `NewDatasourceWithContext` in grafana-aws-sdk, release a new version of grafana-aws-sdk, and use it here. 

Let me know what you all think of this solution for now or if you have another suggestion.

Reply:
I prefer 2, we're going to have to do this for all the aws datasources, right? So it's probably cleaner to do at the `grafana-aws-sdk` level.

Reply:
So we could create a method like the following in grafana-aws-sdk. It still throws away the `ctx`. Any other ideas?

```
func (ds *AsyncAWSDatasource) NewDatasourceWithContext(ctx context.Context, settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
	return ds.NewDatasource(settings)
}
```

In the context of my other comment about getting `ctx` [everywhere properly in grafana-aws-sdk](https://github.com/grafana/athena-datasource/pull/290#discussion_r1377604683), I wonder if it's something that should be done alongside that task.

Reply:
Yeah, it sounds like its part of a bigger refactor. If we do the temporary workaround in 1 we should directly link to it in the refactor issue (and maybe add a comment referencing the issue here) so that we remember to come back to it